samples/templates/express-direct-auth/web-server/routes/login.js

```diff
@@ -1,25 +1,67 @@
 const express = require('express');
 const { getAuthClient } = require('../utils');
+const sampleConfig = require('../../config').webServer;
 
 const router = express.Router();
 
-router.get('/login', (_, res) => {
+const renderLogin = (req, res) => {
   res.render('login');
+};
+
+const renderLoginWithWidget = (req, res) => {
+  const authClient = getAuthClient(req);
+  authClient.idx.interact()
+    .then(idxRes => {
+      const { 
+        interactionHandle, 
+        meta: { codeChallenge, codeChallengeMethod } 
+      } = idxRes;
+      const widgetConfig = JSON.stringify({
+        baseUrl: sampleConfig.oidc.issuer.split('/oauth2')[0],
+        clientId: sampleConfig.oidc.clientId,
+        redirectUri: sampleConfig.oidc.redirectUri,
+        authParams: {
+          issuer: sampleConfig.oidc.issuer,
+          scopes: sampleConfig.oidc.scope.split(' '),
+        },
+        useInteractionCodeFlow: true,
+        interactionHandle,
+        codeChallenge,
+        codeChallengeMethod,
+      });
+      res.render('login-with-widget', {
+        widgetConfig,
+      });
+    })
+    .catch(err => {
+      console.log('Failed to render widget, error: ', err);
+      res.render('login-with-widget', {
+        hasError: true,
+        errors: ['Failed to render widget']
```

Comment:
is this logic implemented? (to show error)

Reply:
Good catch, forgot to add to the template. Generally implemented here:
https://github.com/okta/okta-auth-js/blob/dev-5.0/samples/templates/express-direct-auth/web-server/views/errors.mustache